### PR TITLE
odw.Model

### PR DESCRIPTION
**Pull request by RemDelaporteMathurin**
As discussed in #107 this PR introduces a new `odw.Model` class inheriting from `openmc.Model`.

Since this class stores the geometry, materials, tallies and settings, it makes sense to do all the communication between them in there.

### CODE REVIEW

File: examples/regular_2d_mesh_tally_example.py [hunk 1]
```diff
@@ -79,7 +72,7 @@
 settings.source = FusionRingSource(fuel="DT", radius=350)
 
 # no modifications are made to the default openmc.Model object
-my_model = openmc.Model(
+my_model = odw.Model(
     materials=materials, geometry=geometry, settings=settings, tallies=tallies
 )
 statepoint_file = my_model.run()
```
Comment (RemDelaporteMathurin) [inline examples/regular_2d_mesh_tally_example.py:78]: @shimwell as you can see the line count is not incredibly lower

File: openmc_dagmc_wrapper/model.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    def check_tallies_meshes_corners(self):
+        """Iterates through tallies and check if they have a RegularMesh2D.
+        If the RegularMesh2D doesn't have corners, add them from the geometry.
+        """
+        for tally in self.tallies:
+            for tally_filter in tally.filters:
+                if isinstance(tally_filter, openmc.MeshFilter):
+                    if isinstance(tally_filter.mesh, odw.RegularMesh2D):
+                        if tally_filter.mesh.lower_left is None:
```
Comment (shimwell) [inline openmc_dagmc_wrapper/model.py:40]:
```suggestion
                        if tally_filter.mesh.lower_left is None:
```
Should this also check upper_right

Comment (RemDelaporteMathurin) [reply openmc_dagmc_wrapper/model.py:40]: I wondered... normally if `RegularMesh2D.bounding_box` is `None`, both `lower_left` and `upper_right` are `None`

**Comment by RemDelaporteMathurin** [comment]
@shimwell maybe we should hold this off until the CI is back on?
If I accidentally introduced some bugs here, it will be a nightmare to debug because we are completely blind

**Comment by RemDelaporteMathurin** [comment]
@shimwell I lost track, do we have a repo with the files for the test suite?

**Comment by shimwell** [comment]
Sorry I still don't have a repo with the test files

**Comment by shimwell** [comment]
Getting closer to having a repo with files to download

The last docker release was able to produce files from example 1

https://github.com/fusion-energy/fusion_neutronics_workflow/releases/tag/develop

**Comment by shimwell** [comment]
The h5m files are now being downloaded by the test scripts and the tests are running again. Many are failing but this is something we can work with

**Comment by RemDelaporteMathurin** [comment]
> The h5m files are now being downloaded by the test scripts and the tests are running again. Many are failing but this is something we can work with

Maybe we can work at fixing these tests in develop before merging this. We should try and stop merging PRs when tests don't work :(

**Comment by shimwell** [comment]
Yes sure, I meant work on this PR to fix things up.

**Comment by shimwell** [comment]
Down to 18 failing tests :chart_with_downwards_trend:

**Comment by RemDelaporteMathurin** [comment]
Ah yes `mesh_resolution` is now `resolution` cause `RegularMesh2D` inherits from `openmc.RegularMesh`

**Comment by shimwell** [comment]
@RemDelaporteMathurin I think I've just fixed that bug. Down to just 4 failing tests now

**Comment by shimwell** [comment]
I'm hoping to get this passing all the tests and merged into develop today. Then perhaps we can take a step back and see what parts (if any) of this code are actually useful

**Comment by shimwell** [comment]
@RemDelaporteMathurin the tests are now passing I think this can be merged into the develop branch

**Pull request merged**